tartufo/scanner.py

```diff
@@ -351,30 +350,34 @@ def config_data(self, data: MutableMapping[str, Any]) -> None:
     @cached_property
     def excluded_signatures(self) -> Tuple[str, ...]:
         if self._excluded_signatures is None:
-            signatures = (tuple(self.global_options.exclude_signatures) or ()) + (
-                tuple(self.config_data.get("exclude_signatures", ()))
-            )
-            try:
-                signatures = tuple(
-                    signature["signature"]
-                    for signature in cast(Tuple[Dict[str, str]], signatures)
-                )
-                self._excluded_signatures = tuple(set(signatures))
-                return self._excluded_signatures
-            except TypeError:
+            signatures: Set[str] = set()
+            deprecated = False
+            for signature in tuple(
+                self.global_options.exclude_signatures or []
+            ) + tuple(self.config_data.get("exclude_signatures", [])):
```

Comment:
```pycon
>>> () + []
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate tuple (not "list") to tuple
>>>
```
Careful of the mixture of lists and tuples here. That was part of the cause for the previous type thrashing. 😄

Reply:
That's why both of them are being explicitly converted from lists to tuples here. Also, I played with this extensively last night and python definitely thinks both `self.global_options.exclude_signatures` and `self.config_data["exclude_signatures"]` are lists, which is why I used empty lists for defaults -- to maintain type consistency.

This is strangely bizarre in the sense that `self.global_options.exclude_signatures` pretty clearly appears to be defined as a tuple rather than a list, but if you try removing either of the tuple conversions you get errors complaining that they are lists and that you aren't allowed to concatenate lists this way (or combine a list and a tuple, as the case may be).

I didn't have enough ambition to trace down where the actual content is coming from to understand why mypy isn't whining about some assignment somewhere that is putting a list into a tuple type... Unless it's happening under the covers in click.